### ISSUE TITLE
Rule Change: add `afterHashbangComment` option to `lines-around-comment`

### ISSUE THREAD

**Issue by sosukesuzuki** [eslint/eslint#16913]
### What rule do you want to change?

`lines-around-comment`

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
#!foo
anyStatement;
```

### What does the rule currently do for this code?

Does nothing

### What will the rule do after it's changed?

Warns `Expected line after comment` with following options:

```
"lines-around-comment": ["error", { "afterHashbangComment": true }]
```

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

[Hashbang Comment](https://tc39.es/ecma262/#prod-HashbangComment) is added since ES2023

**Comment by Rec0iL99** [comment]
Hi @sosukesuzuki, thanks for opening the issue. Currently, we are not adding any new options to the `lines-around-comment` rule due to the [stylistic rules policy change](https://github.com/eslint/eslint#stylistic-rule-updates).

But I'm not sure if I should close this, as hashbang comments are added in ES2023. I'll leave this open for someone from the core team to have a better look.

**Comment by mdjermanovic** [comment]
We can consider this option as it adds support for a newly-added ECMAScript feature.

The proposal makes sense to me 👍

**Comment by nzakas** [comment]
No objections from me as it's newly-introduced syntax.
